backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/ElasticsearchRequestLog.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.search.backend.elasticsearch.logging.impl;
+
+import static org.hibernate.search.backend.elasticsearch.logging.impl.ElasticsearchLog.ID_OFFSET_LEGACY_ES;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Map;
+
+import org.hibernate.search.util.common.logging.CategorizedLogger;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+import org.hibernate.search.util.common.logging.impl.MessageConstants;
+
+import org.jboss.logging.BasicLogger;
+import org.jboss.logging.Logger;
+import org.jboss.logging.annotations.LogMessage;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageLogger;
+
+import org.apache.http.HttpHost;
+
+@CategorizedLogger(
+		category = ElasticsearchRequestLog.CATEGORY_NAME,
+		description = """
+				Logs executed requests and responses sent to the Elasticsearch cluster.
+				It also includes the execution time of the request.
+				+
+				Logger must be enabled at TRACE level for this category to print messages.
+				"""
+)
+@MessageLogger(projectCode = MessageConstants.PROJECT_CODE)
+public interface ElasticsearchRequestLog extends BasicLogger {
+	/**
+	 * This is the category of the Logger used to print out executed Elasticsearch requests,
+	 * along with the execution time.
+	 * <p>
+	 * To enable the logger, the category needs to be enabled at TRACE level.
+	 */
+	String CATEGORY_NAME = "org.hibernate.search.backend.elasticsearch.request";
```

Comment:
I was going back and forth with this one 🙈 we also already had a `org.hibernate.search.backend.lucene.infostream` that had backend in it ... so I was like.. add a backend to the Elasticsearch category ... drop `.backend` from the infostream one? It felt inconsistent to keep it as is 🙈 

> so that people can just enable debug of the client and automatically get request logging?

I'll check if it'll work like that... I thought it only works for "exact match" but even if it doesn't the suggestion still makes sense to move request part after client 👍🏻